test/libyul/yulOptimizerTests/equalStoreEliminator/mstore8_mstore8.yul

```diff
@@ -0,0 +1,21 @@
+{
+    let _0 := 0
+    let _31 := 31
+
+    calldatacopy(0, 0, 0x40)
+    let mem31 := mload(_31)
+
+    mstore8(_0, mem31)
+    mstore8(_0, mem31) // Redundant with previous MSTORE8.
+}
+// ----
+// step: equalStoreEliminator
+//
+// {
+//     let _0 := 0
+//     let _31 := 31
+//     calldatacopy(0, 0, 0x40)
+//     let mem31 := mload(_31)
+//     mstore8(_0, mem31)
```

Comment:
I am now confused - this is an `mstore`, so it should be removed, no? How does `mcopy` affect this?

Reply:
Ah, wait, it's *not* getting removed here :) Only `MSTORE` in the other test gets removed.

Yeah, that's part of the reason why I added tests for `MSTORE8` as well. I think this optimizer step only considers full slots/words. But I did not investigate deeper. I doubt it's a bug. Looks more like a limitation of the step. In any case, the code it produces is not broken, just not as well optimized as it could :)

Reply:
Yup, it looks like a limitation of the optimizer step, but as you said, it produces valid code, so it's fine.